examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -424,7 +465,7 @@ export class TaskList extends DataObject implements ITaskList {
  * scenario, the fourth argument is not used.
  */
 export const TaskListInstantiationFactory = new DataObjectFactory<TaskList>(
-	"task-list",
+	"task-list-1",
```

Comment:
Here you probably just want `"task-list"` - this is just used for identifying the TaskList class, not a particular instance.

Reply:
Hmm I see -- yes what I want to do is to identify a task list by a unique id that the external data service will provide, and then be able to instantiate and identify it within the Fluid ecosystem with that same ID. At the moment, it's unclear to me how to accomplish that given that taskList instantiation is so tied to container instantiation.

Reply:
I think you hit the nail on the head with that last sentence. We probably want to decouple tasklist creation from container instantiation, which probably is more realistic (so a container can support having different task lists). *Some* data-object/data-store will still get instantiated at the root of the container when it instantiates, but that can be a "task list collection" (initially empty) like Matt suggested in another comment.

I'll shamelessly plug in the entryPoint-in-container work that got merged into main recently, to say that maybe we could make the functionality to create a new task list (passing in the external id) a method on the container's entryPoint, and the app would do something like

```typescript
const entryPoint = await container.getEntryPoint?.();
entryPoint?.createTaskList(<task-list-id>);
```

Entrypoint is not strongly typed, so I'm just a bit conflicted between telling eslint to be quiet or using `as ...` casts.

Reply:
Just bear in mind that you'll also want additional functionality like `.getTaskLists()` and `.getTaskList(id)`, since the host (which is invoking the entryPoint) won't know by default the IDs of the task lists inside or how to retrieve them.  This probably points to a need for a real DataObject so you can persist the list of IDs and/or collection of TaskLists, though there can be other approaches.